backend/testfiles/execution/string.tests

```diff
@@ -577,6 +577,12 @@ String.padEnd_v0 "🧟‍♀️🧟‍♂️" "0" 5 = Ok "🧟‍♀️🧟‍
 String.padEnd_v0 "👨‍❤️‍💋‍👨👩‍👩‍👧‍👦🏳️‍⚧️🇵🇷" "0" 10 = Ok "👨‍❤️‍💋‍👨👩‍👩‍👧‍👦🏳️‍⚧️🇵🇷000000"
 String.padEnd_v0 "鷝" "觌഻" 0 = Ok "鷝"
 
+String.indexOf_v0 "hello world" "world" = 6
+String.indexOf_v0 "hello world" "earth" = -1
+String.indexOf_v0 "" "" = 0
+String.indexOf_v0 "hello" "" = 0
+String.indexOf_v0 "" "hello" = -1
```

Comment:
it might be worth adding some more complicated tests involving strings of more complicated Unicode characters - for example, borrowing from String.split tests, `String.indexOf "👨‍👩‍👧‍👦" "👩" = None`.

OK to leave this as a CLEANUP item

Reply:
(worth checking out String.split tests generally, for inspiration)